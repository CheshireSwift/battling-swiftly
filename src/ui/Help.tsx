import * as React from 'react'
import * as _ from 'lodash'

import { cMenuBg } from '../styling/constants'
import Options, { SetOption, initialOptions } from '../data/Options'
import customColorOptions from "../graphics/customColorOptions";

export const Help = ({
  show,
  setOption,
  onClose,
}: {
  show: boolean
  setOption: SetOption
  onClose: () => void
}) => {
  const options = React.useContext(Options)
  return show ? (
    <div
      style={{
        position: 'fixed',
        top: '10%',
        left: '25%',
        width: '50%',
        color: options.drawColour,
        background: cMenuBg,
        border: `1px solid ${options.drawColour}`,
        padding: '0 2rem',
        fontSize: 18,
      }}
    >
      <h1 style={{ textAlign: 'center' }}>Help and Options</h1>
      <button
        style={{
          position: 'absolute',
          top: -1,
          right: -1,
          border: `1px solid ${options.drawColour}`,
          background: 'none',
          color: options.drawColour,
          fontSize: 18,
          padding: '0.1rem',
        }}
        onClick={onClose}
      >
        â•³
      </button>
      <ul>
        <li>
          Scale
          <input
            style={{
              margin: '1rem',
              background: 'black',
              border: `1px solid ${options.drawColour}`,
              color: options.drawColour,
              fontFamily: 'monospace',
            }}
            type="number"
            value={
              _.isFinite(options.fixedScale)
                ? (options.fixedScale as number)
                : ''
            }
            min="0"
            max="5"
            step="0.5"
            onChange={e => setOption('fixedScale', e.target.valueAsNumber)}
            placeholder="Variable"
          />
        </li>
        <li>
          Plotting colour
          <select
            style={{
              margin: '1rem',
              background: 'black',
              border: `1px solid ${options.drawColour}`,
              color: options.drawColour,
              fontFamily: 'monospace',
            }}
            value={options.drawColour}
            onChange={e =>
              setOption(
                'drawColour',
                e.target.value || initialOptions.drawColour,
              )
            }
          >
            {_.map(Object.entries(customColorOptions), ([key, value]) =>
              (
                <option
                  key={key}
                  value={value}
                  style={{
                    color: `${value}`
                  }}
                >
                  {key}
                </option>
              )
            )}
          </select>
        </li>
      </ul>
    </div>
  ) : null
}

export default Help
