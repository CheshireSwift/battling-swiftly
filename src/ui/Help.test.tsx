import * as React from 'react'
import { render } from 'enzyme'

import Help from './Help'
describe('the help display', () => {
  it('matches the snapshot', () => {
    expect(
      render(<Help show={true} setOption={jest.fn()} onClose={jest.fn()} />),
    ).toMatchSnapshot()
  })

  it('hides when not shown', () => {
    expect(
      render(<Help show={false} setOption={jest.fn()} onClose={jest.fn()} />)
        .length,
    ).toBe(0)
  })

  it('displays some content when shown', () => {
    expect(
      render(<Help show={true} setOption={jest.fn()} onClose={jest.fn()} />)
        .length,
    ).toBeGreaterThan(0)
  })
})
