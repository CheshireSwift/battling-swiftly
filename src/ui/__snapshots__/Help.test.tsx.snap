// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the help display matches the snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "position:fixed;top:10%;left:25%;width:50%;color:lime;background:#000000dd;border:1px solid lime;padding:0 2rem;font-size:18px",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "style": "text-align:center",
        },
        "children": Array [
          Object {
            "data": "Help and Options",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h1",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "style": "position:absolute;top:-1px;right:-1px;border:1px solid lime;background:none;color:lime;font-size:18px;padding:0.1rem",
          },
          "children": Array [
            Object {
              "data": "╳",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "Scale",
                    "next": Object {
                      "attribs": Object {
                        "max": "5",
                        "min": "0",
                        "placeholder": "Variable",
                        "step": "0.5",
                        "style": "margin:1rem;background:black;border:1px solid lime;color:lime;font-family:monospace",
                        "type": "number",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "max": undefined,
                        "min": undefined,
                        "placeholder": undefined,
                        "step": undefined,
                        "style": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "max": undefined,
                        "min": undefined,
                        "placeholder": undefined,
                        "step": undefined,
                        "style": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                  Object {
                    "attribs": Object {
                      "max": "5",
                      "min": "0",
                      "placeholder": "Variable",
                      "step": "0.5",
                      "style": "margin:1rem;background:black;border:1px solid lime;color:lime;font-family:monospace",
                      "type": "number",
                      "value": "",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "data": "Scale",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "max": undefined,
                      "min": undefined,
                      "placeholder": undefined,
                      "step": undefined,
                      "style": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "max": undefined,
                      "min": undefined,
                      "placeholder": undefined,
                      "step": undefined,
                      "style": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                ],
                "name": "li",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
            ],
            "name": "ul",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "style": "position:absolute;top:-1px;right:-1px;border:1px solid lime;background:none;color:lime;font-size:18px;padding:0.1rem",
        },
        "children": Array [
          Object {
            "data": "╳",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "Scale",
                  "next": Object {
                    "attribs": Object {
                      "max": "5",
                      "min": "0",
                      "placeholder": "Variable",
                      "step": "0.5",
                      "style": "margin:1rem;background:black;border:1px solid lime;color:lime;font-family:monospace",
                      "type": "number",
                      "value": "",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "max": undefined,
                      "min": undefined,
                      "placeholder": undefined,
                      "step": undefined,
                      "style": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "max": undefined,
                      "min": undefined,
                      "placeholder": undefined,
                      "step": undefined,
                      "style": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
                Object {
                  "attribs": Object {
                    "max": "5",
                    "min": "0",
                    "placeholder": "Variable",
                    "step": "0.5",
                    "style": "margin:1rem;background:black;border:1px solid lime;color:lime;font-family:monospace",
                    "type": "number",
                    "value": "",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "data": "Scale",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "max": undefined,
                    "min": undefined,
                    "placeholder": undefined,
                    "step": undefined,
                    "style": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "max": undefined,
                    "min": undefined,
                    "placeholder": undefined,
                    "step": undefined,
                    "style": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "ul",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "style": "text-align:center",
          },
          "children": Array [
            Object {
              "data": "Help and Options",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h1",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "Scale",
                "next": Object {
                  "attribs": Object {
                    "max": "5",
                    "min": "0",
                    "placeholder": "Variable",
                    "step": "0.5",
                    "style": "margin:1rem;background:black;border:1px solid lime;color:lime;font-family:monospace",
                    "type": "number",
                    "value": "",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "max": undefined,
                    "min": undefined,
                    "placeholder": undefined,
                    "step": undefined,
                    "style": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "max": undefined,
                    "min": undefined,
                    "placeholder": undefined,
                    "step": undefined,
                    "style": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
              Object {
                "attribs": Object {
                  "max": "5",
                  "min": "0",
                  "placeholder": "Variable",
                  "step": "0.5",
                  "style": "margin:1rem;background:black;border:1px solid lime;color:lime;font-family:monospace",
                  "type": "number",
                  "value": "",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "data": "Scale",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "max": undefined,
                  "min": undefined,
                  "placeholder": undefined,
                  "step": undefined,
                  "style": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "max": undefined,
                  "min": undefined,
                  "placeholder": undefined,
                  "step": undefined,
                  "style": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "ul",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "style": "position:absolute;top:-1px;right:-1px;border:1px solid lime;background:none;color:lime;font-size:18px;padding:0.1rem",
          },
          "children": Array [
            Object {
              "data": "╳",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "style": "text-align:center",
            },
            "children": Array [
              Object {
                "data": "Help and Options",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h1",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
